apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js

```diff
@@ -8,6 +8,99 @@ window.PLOrderBlocks = function (uuid, options) {
+      let parentArea = block.closest('.pl-order-blocks-connected-sortable');
+      return parentArea.classList.contains('dropzone');
+    }
+    function getIndentation(block) {
+      return Math.round(parseInt(block.style.marginLeft.replace('px', '') / TABWIDTH));
+    }
+    let fullContainer = document.querySelector('.pl-order-blocks-question-' + uuid);
+    let blocks = fullContainer.querySelectorAll('*.pl-order-block');
+
+    for (let block of blocks) {
+      block.addEventListener('click', () => {
+        block.focus();
+      });
+
+      block.addEventListener('blur', () => {
+        if (block.classList.contains('selected')) {
+          block.classList.remove('selected');
+        }
+      });
+
+      block.addEventListener('keydown', (ev) => {
+        pressedKeys[ev.key] = true;
+        if (!block.classList.contains('selected')) {
+          if (ev.key === 'Enter') {
+            block.classList.add('selected');
+          }
+        } else {
+          ev.preventDefault();
+          switch (ev.key) {
+            case 'ArrowDown':
+              if (block.nextElementSibling) {
+                block.nextElementSibling.insertAdjacentElement('afterend', block);
+              }
+              break;
+            case 'ArrowUp':
+              if (block.previousElementSibling) {
+                block.previousElementSibling.insertAdjacentElement('beforebegin', block);
+              }
+              break;
+            case 'Delete':
+            case 'Backspace':
+              if (inDropzone(block)) {
+                block.style.marginLeft = '0px';
+                if (block.hasAttribute('data-distractor-bin')) {
+                  let distractorBinId = block.getAttribute('data-distractor-bin');
+                  let distractorBin = fullContainer
+                    .querySelector(
+                      '.pl-order-blocks-pairing-indicator[data-distractor-bin=' +
+                        distractorBinId +
+                        ']',
+                    )
+                    .querySelector('.inner-list');
+                  distractorBin.insertAdjacentElement('afterbegin', block);
+                } else {
+                  $(optionsElementId)[0].insertAdjacentElement('afterbegin', block);
+                }
+              }
+              break;
+            case 'Enter':
+              if (!inDropzone(block)) {
+                $(dropzoneElementId)[0].insertAdjacentElement('afterbegin', block);
+              }
+              break;
+            case 'Tab':
+              if (inDropzone(block) && enableIndentation) {
+                let currentIndent = getIndentation(block);
+                if (pressedKeys['Shift'] && currentIndent > 0) {
+                  block.style.marginLeft = (currentIndent - 1) * TABWIDTH + 'px';
+                } else if (currentIndent < maxIndent && !pressedKeys['Shift']) {
+                  block.style.marginLeft = (currentIndent + 1) * TABWIDTH + 'px';
+                }
```

Comment:
this indentation calculation should be moved to a function so it doesn't need to be kept the same between multiple places in the code